File: DSA/done/217 - containsDuplicate.cpp
// https://leetcode.com/problems/contains-duplicate/

#include <unordered_set>
#include <vector>

using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> seen;
        for (int n : nums) {
            if (seen.count(n) > 0) {
                return true;
            }
            seen.insert(n);
        }
        return false;
    }
};

File: DSA/done/217 - containsDuplicate.py
# https://leetcode.com/problems/contains-duplicate/

from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hm = set()
        for n in nums:
            if n in hm:
                return True
            hm.add(n)
        return False